File: Practice/src/introduction_coding_tests/CharacterCoordinates.js
function solution(keyinput, board) {
    var answer = [0, 0];
    const d = { left: [-1, 0], right: [1, 0], up: [0, 1], down: [0, -1] };
    for (let i of keyinput) {
        let x = answer[0] + d[i][0];
        let y = answer[1] + d[i][1];
        if (Math.abs(x) <= board[0] / 2 && Math.abs(y) <= board[1] / 2) {
            answer[0] = x;
            answer[1] = y;
        }
    }
    return answer;
}

File: Practice/src/introduction_coding_tests/CharacterCoordinates.py
def solution(keyinput, board):
    answer = [0, 0]
    d = {"left": [-1, 0], "right": [1, 0], "up": [0, 1], "down": [0, -1]}
    for i in keyinput:
        if abs(answer[0] + d[i][0]) <= board[0] / 2 and abs(answer[1] + d[i][1]) <= board[1] / 2:
            answer[0] += d[i][0]
            answer[1] += d[i][1]
    return answer
